### ISSUE TITLE
[NEWS] Autoryzacja - wydłużyliśmy ważność refresh tokenów do 60 sekund po pierwszym użyciu / Authorization - we have extended refresh tokens validity to 60 seconds after first use

### ISSUE THREAD

**Issue by PrzemyslawLukanowski** [allegro/allegro-api#4136]
Refresh tokeny były do tej pory jednorazowego użytku - w momencie wygenerowania nowej pary (access i refresh token) stare tokeny od razu traciły ważność. **Wydłużyliśmy żywotność refresh tokenów do 60 sekund od momentu odświeżenia**. Chcemy w ten sposób rozwiązać problem utraconych refresh tokenów na skutek przerwanego połączenia. W takiej sytuacji jedynym rozwiązaniem było ponowne przejście całego procesu autoryzacji użytkownika w celu wygenerowania tokena dostępu i refresh tokena. Obecnie nie jest to konieczne, możesz ponowić próbę odświeżenia tokena za pomocą tego samego refresh tokena. 

-----

Refresh tokens have so far been disposable - after generating a new pair (access and refresh token), the previous ones immediately expired. **We extended the validity of refresh tokens to 60 seconds after first use**. Owing to it we want to solve the problem of lost refresh tokens due to interrupted connection. In such a situation, the only solution was to go through the entire user authorization process again to generate an access token and a refresh token. At present, it is not necessarily - you can retry to refresh the token with the same refresh token.

**Comment by Onixarts** [comment]
To pytanko mam, jakie tokeny będą zwrócone przy dwóch próbach odświeżenia z użyciem starego tokena? Zakładając, że robią to dwa niezależne procesy, pierwszy odświeży, otrzyma nową parę tokenów. Jakie tokeny otrzyma drugi proces, korzystając z tego już użytego refresh tokena? Te same co pierwszy czy kolejne nowe?

**Comment by PrzemyslawLukanowski** [comment]
Otrzymasz nowe tokeny.

**Comment by Onixarts** [comment]
Ok, będziemy obserwować czy to coś dało. P.S. czy na sandboxie też to działa?

**Comment by PrzemyslawLukanowski** [comment]
Tak, na sandboxie również refresh tokeny są ważne przez 60s po pierwszym odświeżeniu.
